File: src/api/payment.ts
import api from './config';
import type { ApiResponse } from '../types';

export interface Payment {
  id: number;
  paymentSn: string;
  orderSn: string;
  payType: number;
  amount: number;
  status: number;
  createTime: string;
  payTime?: string;
}

export const createPayment = (data: {
  orderSn: string;
  payType: number;
  amount: number;
}): Promise<ApiResponse<Payment>> => {
  return api.post('/api/payment/create', data);
};

export const getPaymentStatus = (paymentSn: string): Promise<ApiResponse<Payment>> => {
  return api.get(`/api/payment/status/${paymentSn}`);
};

export const paymentCallback = (paymentSn: string): Promise<ApiResponse<Payment>> => {
  return api.post(`/api/payment/callback/${paymentSn}`);
};

File: mall-parent/mall-payment/src/main/java/com/mall/payment/entity/Payment.java
package com.mall.payment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("pms_payment")
public class Payment {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String orderSn;

    private String paymentSn;

    private Long userId;

    private BigDecimal amount;

    private Integer payType;

    private Integer status;

    private LocalDateTime createTime;

    private LocalDateTime payTime;
}
